chef/automate
# Dan/manual svc delete cli

### :nut_and_bolt: Description: What code changed, and why?

Adds the necessary API to support the manual service deletion flow described in #1192 and provides a CLI to use it.

### :chains: Related Resources

#1192
#2488 

### :athletic_shoe: How to Build and Test the Change

* *Build:* automate-gateway, automate-cli, applications-service
* Start all services and run `applications_populate_database`
* `chef-automate applications show-svcs` shows all the services
* `chef-automate applications show-svcs -a "*" -b "*" -c "*" -e "*" -g "*" -o "*" -n "*" -s "*" -v "*" -D` uses all the filtering options (this will probably not return any services unless you waited for them to become disconnected)
* Experiment with any of the filter options (see what they mean with `--help`)
* Delete services with `chef-automate applications remove-svcs`, they should not show up in the `show-svcs` command after.

### :white_check_mark: Checklist


### :camera: Screenshots, if applicable

```
chef-automate applications show-svcs -a "*" -b "*" -c "*" -e "*" -g "*" -o "*" -n "*" -s "*" -v "*" | head -5
id   	svc.group    	release                          	FQDN                	app:env  	status  
17741	redis.default	custom/redis/1.0.1/20190115189830	ebnru2er.example.net	demo:demo	CRITICAL
17695	nginx.default	custom/nginx/2.0.0/20190115189976	7810o1k8.example.net	demo:demo	CRITICAL
17734	redis.default	custom/redis/1.0.1/20190115189830	djfjxnrq.example.net	demo:demo	CRITICAL
17676	nginx.default	custom/nginx/2.0.0/20190115189976	23hacbg2.example.net	demo:demo	CRITICAL
17670	billing-svc.default	custom/billing-svc/1.3.4/20190115182382	jb0dbg8y.example.net	billing-backend:qa	CRITICAL
```

```
[258][default:/src:0]# chef-automate applications remove-svcs -a "m*" -b "*" -c "*" -e "p*" -g "*" -o "*" -n "mobile-api-auth*" -s "*" -v "*" 
id   	svc.group              	release                                    	FQDN                	app:env                  	status
18542	mobile-api-auth.default	custom/mobile-api-auth/2.0.8/20190115181192	bllhcayc.example.net	mobile-app-api:production	OK

The above 1 services will be deleted. Do you wish to continue? (y/n)
y
Removed 1 services
[259][default:/src:0]# chef-automate applications show-svcs -a "m*" -b "*" -c "*" -e "p*" -g "*" -o "*" -n "mobile-api-auth*" -s "*" -v "*" 
id	svc.group	release	FQDN	app:env	status
```
